de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/JsonHandler.java

```diff
@@ -36,108 +34,97 @@
 
   protected ScriptHandler scriptHandler;
   protected ExecutionContext exec;
-  
+  protected ParameterData parameterJson;
+  protected ObjectMapper MAPPER = new ObjectMapper();
+
   protected JsonHandler(ScriptHandler scriptHandler, ExecutionContext exec) {
     this.scriptHandler = scriptHandler;
     this.exec = exec;
-    
+    this.parameterJson = new ParameterData();
     try {
       importLibraries();
-    } catch(Exception e) {
+    } catch (Exception e) {
       e.printStackTrace();
     }
   }
-  
+
   // TODO: conversion command? maybe create a new Class instead of using a map
-  public void applyJoinRelation(FskPortObject fskObj,
-      List<JoinRelationAdvanced> joinRelationList,
+  public void applyJoinRelation(FskPortObject fskObj, List<JoinRelationAdvanced> joinRelationList,
       String suffix) throws Exception {
-    
-      if(joinRelationList != null ) {
-        for (JoinRelationAdvanced joinRelation: joinRelationList) {
-          String targetParameter = joinRelation.getTargetParam();
-          FskPortObject sourceModel = joinRelation.getModel();
-          String sourceParameter = joinRelation.getSourceParam();
-          
-          for (Parameter param : SwaggerUtil.getParameter(fskObj.modelMetadata)) {
-            if (targetParameter.equals(param.getId() + suffix)) {
-              String resourcePath = sourceModel.getGeneratedResourcesDirectory().get().getAbsolutePath()
-                  .replaceAll("\\\\", "/") + "/";
-              String jsonPath = resourcePath + JSON_FILE_NAME;
-              loadParametersIntoWorkspace(jsonPath, sourceParameter, param.getId());
-              
-           // if target parameter is of type FILE, add path to generatedResources to sourceParam
-              // This should be safe since source and target parameter must have the same type (if File)
-              if(param.getDataType().equals(Parameter.DataTypeEnum.FILE)) {
-                addPathToFileParameter(param.getId(), resourcePath);
-              }
-              String rep = joinRelation.replaceCommand(param.getId());
-              applyJoinCommand(param.getId(), joinRelation.replaceCommand(param.getId()));
+
+    if (joinRelationList != null) {
+      for (JoinRelationAdvanced joinRelation : joinRelationList) {
+        String targetParameter = joinRelation.getTargetParam();
+        FskPortObject sourceModel = joinRelation.getModel();
+        String sourceParameter = joinRelation.getSourceParam();
+
+        for (Parameter param : SwaggerUtil.getParameter(fskObj.modelMetadata)) {
+          if (targetParameter.equals(param.getId() + suffix)) {
+            String resourcePath = sourceModel.getGeneratedResourcesDirectory().get()

```

Comment:
generated resources directory is optional and may be missing. We should check ifPresent.